cmd/goal/clerk.go

```diff
@@ -724,46 +724,77 @@ var signCmd = &cobra.Command{
 
 		var outData []byte
 		dec := protocol.NewDecoderBytes(data)
+		// read the entire file and prepare in-memory copy of each signed transaction, with grouping.
+		txnGroups := make(map[crypto.Digest][]*transactions.SignedTxn)
```

Comment:
nit: It is OK to use obj copies instead of pointers in CLI code (imo code without pointers looks cleaner, and this is a CLI so mem efficiency has low impact)

Reply:
I think that you missed the point of me using pointers... Pointers are unique, whereas `transactions.SignedTxn` might not be. If the input has the same transaction multiple times, and we load it all into a map and then "dump" it, we would skip duplicate transactions. Another reason is that it allows me to "tag" each transaction with it's index within the input file, which could be problematic if we were using `transactions.SignedTxn`.

Reply:
ah, I see it is really only needed for  proper index displaying `txnIndex[txnGroups[group][i]]` in the error message.